### PR TITLE
[runtime] Grab bag of MonoError fixes in object.c

### PR DESCRIPTION
**Pull request by lambdageek**
Largest change is `mono_array_new` → `mono_array_new_checked`

### CODE REVIEW

File: mono/metadata/socket-io.c [hunk 1]
```diff
@@ -1924,11 +1924,10 @@ ves_icall_System_Net_Sockets_Socket_Select_internal (MonoArray **sockets, gint32
 }
 
 static MonoObject*
-int_to_object (MonoDomain *domain, int val)
+int_to_object (MonoDomain *domain, int val, MonoError *error)
 {
-	MonoError error;
-	MonoObject *result = mono_value_box_checked (domain, mono_get_int32_class (), &val, &error);
-	mono_error_raise_exception (&error); /* FIXME don't raise here */
+	mono_error_init (error);
```
Comment (ludovic-henry) [inline mono/metadata/socket-io.c]: we don't need to init `error` here.

File: mono/metadata/object.c [hunk 2]
```diff
@@ -5346,7 +5348,29 @@ mono_array_clone (MonoArray *array)
...
+ * Returns: A newly created array who is a shallow copy of @array.  On
+ * failure returns NULL and sets @error.
+ */
+MonoArray*
+mono_array_clone_checked (MonoArray *array, MonoError *error)
+{
+
+	MONO_REQ_GC_UNSAFE_MODE;
+
+	mono_error_init (error);
```
Comment (ludovic-henry) [inline mono/metadata/object.c]: you don't need to init the error here

File: mono/metadata/appdomain.c [hunk 3]
```diff
@@ -2210,7 +2212,8 @@ ves_icall_System_AppDomain_ExecuteAssembly (MonoAppDomain *ad,
 		g_error ("No entry point method found in %s due to %s", image->name, mono_error_get_message (&error));
 
 	if (!args)
-		args = (MonoArray *) mono_array_new (ad->data, mono_defaults.string_class, 0);
+		args = (MonoArray *) mono_array_new_checked (ad->data, mono_defaults.string_class, 0, &error);
+	mono_error_assert_ok (&error);
```
Comment (ludovic-henry) [inline mono/metadata/appdomain.c]: is it on purpose that it's not in the same block of code as the previous line?

File: mono/metadata/icall.c [hunk 4]
```diff
@@ -2692,10 +2707,11 @@ ves_icall_RuntimeTypeHandle_GetArrayRank (MonoReflectionType *type)
 }
 
 static MonoArray*
-create_type_array (MonoDomain *domain, MonoBoolean runtimeTypeArray, int count)
+create_type_array (MonoDomain *domain, MonoBoolean runtimeTypeArray, int count, MonoError *error)
 {
 	MonoArray *res;
-	res = mono_array_new (domain, runtimeTypeArray ? mono_defaults.runtimetype_class : mono_defaults.systemtype_class, count);
+	mono_error_init (error);
```
Comment (ludovic-henry) [inline mono/metadata/icall.c]: no need to init error here

File: mono/metadata/locales.c [hunk 5]
```diff
@@ -181,31 +191,70 @@ ves_icall_System_Globalization_CalendarData_fill_calendar_data (MonoCalendarData
...
 
 	MONO_OBJECT_SETREF (this_obj, NativeName, mono_string_new (domain, idx2string (ci->nativename)));
-	MONO_OBJECT_SETREF (this_obj, ShortDatePatterns, create_names_array_idx_dynamic (dfe->short_date_patterns,
-			NUM_SHORT_DATE_PATTERNS));
-	MONO_OBJECT_SETREF (this_obj, YearMonthPatterns, create_names_array_idx_dynamic (dfe->year_month_patterns,
-			NUM_YEAR_MONTH_PATTERNS));
+	MonoArray *short_date_patterns = create_names_array_idx_dynamic (dfe->short_date_patterns,
+									 NUM_SHORT_DATE_PATTERNS, &error);
+	if (mono_error_set_pending_exception (&error))
+		return FALSE;
```
Comment (ludovic-henry) [inline mono/metadata/locales.c]: you could create a macro for that

**Comment by lambdageek** [comment]
Thanks @ludovic-henry.  I revised the code to incorporate your comments.

**Pull request merged**